src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -312,6 +325,58 @@ private void waitForSecurityIndexToBeAtLeastYellow() {
+                    LOGGER.info("Creating index {} and default configs if they are absent", securityIndex);
+                    createSecurityIndexIfAbsent();
+                    uploadConfigurationFiles();
+                    loadConfiguration("Init default configuration on manager node", CType.values());
+                }
+                return true;
+            } catch (Exception e) {
+                LOGGER.error("Cannot apply default config (this is maybe not an error!)", e);
+                return false;
+            }
+        });
+    }
+
+    private void uploadConfigurationFiles() {
+        try {
+            final var configDir = resolveConfigDir();
+            if (Files.exists(CType.CONFIG.configFile(configDir))) {
+                final ThreadContext threadContext = threadPool.getThreadContext();
+                try (StoredContext ctx = threadContext.stashContext()) {
+                    threadContext.putHeader(ConfigConstants.OPENDISTRO_SECURITY_CONF_REQUEST_HEADER, "true");
+                    ConfigHelper.uploadFile(client, CType.CONFIG, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ROLES, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ROLESMAPPING, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.INTERNALUSERS, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    ConfigHelper.uploadFile(client, CType.ACTIONGROUPS, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    // FIXME remove it as soon as we will read of version 1 config
+                    if (DEFAULT_CONFIG_VERSION == 2) {
+                        ConfigHelper.uploadFile(client, CType.TENANTS, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    }
+                    final boolean emptyIfFileMissing = true;
+                    ConfigHelper.uploadFile(client, CType.NODESDN, configDir, securityIndex, DEFAULT_CONFIG_VERSION, emptyIfFileMissing);
+                    ConfigHelper.uploadFile(client, CType.WHITELIST, configDir, securityIndex, DEFAULT_CONFIG_VERSION, emptyIfFileMissing);
+                    ConfigHelper.uploadFile(client, CType.ALLOWLIST, configDir, securityIndex, DEFAULT_CONFIG_VERSION, emptyIfFileMissing);
+                    // Audit config is not packaged by default
+                    if (Files.exists(CType.AUDIT.configFile(configDir))) {
+                        ConfigHelper.uploadFile(client, CType.AUDIT, configDir, securityIndex, DEFAULT_CONFIG_VERSION);
+                    }
+                }
+            } else {
+                LOGGER.error("{} does not exist", CType.CONFIG.configFile(configDir).toAbsolutePath());
+            }
+        } catch (Exception e) {
+            LOGGER.error("Cannot apply default config (this is maybe not an error!)", e);
```

Comment:
`(this is maybe not an error!)` Can we remove this, or handle non-error cases better?

Reply:
sure I will